diff --exclude '*~' -ruN A/doom-modeline.el B/doom-modeline.el
--- A/doom-modeline.el	2020-01-13 15:58:28.439645863 -0500
+++ B/doom-modeline.el	2020-01-13 16:03:03.532895369 -0500
@@ -102,7 +102,7 @@
   '(objed-state misc-info battery irc-buffers debug minor-modes input-method indent-info buffer-encoding major-mode process))
 
 (doom-modeline-def-modeline 'project
-  '(bar window-number buffer-default-directory)
+  '(bar window-number modals buffer-default-directory)
   '(misc-info battery irc mu4e gnus github debug major-mode process))
 
 (doom-modeline-def-modeline 'vcs
