diff -ruN --exclude '*~' A/doom-modeline-segments.el B/doom-modeline-segments.el
--- A/doom-modeline-segments.el	2019-11-19 11:56:34.927335789 -0500
+++ B/doom-modeline-segments.el	2019-11-19 12:09:45.564497276 -0500
@@ -1512,10 +1512,16 @@
 
 (defsubst doom-modeline--god ()
   "The current god state. Requires `god-mode' to be enabled."
-  (when (bound-and-true-p god-local-mode)
-    (propertize " <G> " 'face (if (doom-modeline--active)
-                                  'doom-modeline-evil-normal-state
-                                'mode-line-inactive))))
+  (when (boundp god-local-mode)
+    ;; (propertize " <G> " 'face (if (doom-modeline--active)
+    ;;                               'doom-modeline-evil-normal-state
+    ;;                             'mode-line-inactive))
+    (all-the-icons-faicon "circle" :face (if (doom-modeline--active)
+                                             (if god-local-mode
+                                                 'all-the-icons-green
+                                               'all-the-icons-red)
+                                           'all-the-icons-silver))
+    ))
 
 (defsubst doom-modeline--ryo ()
   "The current ryo-modal state. Requires `ryo-modal-mode' to be enabled."
diff -ruN --exclude '*~' A/doom-modeline.el B/doom-modeline.el
--- A/doom-modeline.el	2019-11-19 11:56:34.928335790 -0500
+++ B/doom-modeline.el	2019-11-19 11:57:04.191379365 -0500
@@ -101,7 +101,7 @@
   '(objed-state misc-info battery irc-buffers debug lsp minor-modes input-method indent-info buffer-encoding major-mode process checker))
 
 (doom-modeline-def-modeline 'project
-  '(bar window-number buffer-default-directory)
+  '(bar window-number modals buffer-default-directory)
   '(misc-info battery mu4e github debug major-mode process))
 
 (doom-modeline-def-modeline 'package
