diff -ruN --exclude '*~' A/deft.el B/deft.el
--- A/deft.el	2021-07-13 16:09:32.411560358 -0400
+++ B/deft.el	2021-07-13 16:20:00.077925185 -0400
@@ -548,6 +548,7 @@
 
 (require 'cl-lib)
 (require 'button)
+(require 'org-roam)
 
 ;; Customization
 
@@ -1084,10 +1085,7 @@
 used as title."
   (if deft-use-filename-as-title
       (deft-base-filename file)
-    (let ((begin (string-match "^.+$" contents)))
-      (if begin
-          (funcall deft-parse-title-function
-                   (substring contents begin (match-end 0)))))))
+    (org-roam-db--get-title file)))
 
 (defun deft-parse-summary (contents title)
   "Parse the file CONTENTS, given the TITLE, and extract a summary.
