diff -Naur old/source/wscript new/source/wscript
--- source/wscript	1969-12-31 19:00:01.000000000 -0500
+++ source/wscript	2018-01-05 12:34:13.000000000 -0500
@@ -293,22 +293,8 @@
     lib.append('ncurses')
 
   if bld.env['use_system_clang']:
-    if bld.env['llvm_config']:
-      # If --llvm-config is specified, set RPATH and use $bindir/bin/clang -###
-      # to detect recource directory.
-      output = str(subprocess.check_output(
-          [bld.env['llvm_config'], '--bindir'],
-          stderr=subprocess.STDOUT).decode()).strip()
-      clang = os.path.join(output, 'clang')
-      rpath = str(subprocess.check_output(
-          [bld.env['llvm_config'], '--libdir'],
-          stderr=subprocess.STDOUT).decode()).strip()
-    else:
-      clang = 'clang'
-      if sys.platform == 'darwin':
-        rpath = bld.env['LIBPATH_clang'][0]
-      else:
-        rpath = []
+    clang = 'clang'
+    rpath = []
 
     devnull = '/dev/null' if sys.platform != 'win32' else 'NUL'
     output = subprocess.check_output(
